includes/tasks/class-wc-calypso-task-headstart-products.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks;
+
+use Automattic\WooCommerce\Admin\Features\Features;
+use Automattic\WooCommerce\Admin\Features\OnboardingTasks\Tasks\Products;
+
+/**
+ * HeadstartProducts Task
+ */
+class HeadstartProducts extends Products {
+	/**
+	 * Task completion.
+	 *
+	 * @return bool
+	 */
+	public function is_complete() {
+		return self::has_products();
+	}
+
+	/**
+	 * Check if the store has any published products,
+	 * but exclude sample products generated by Headstart themes.
+	 *
+	 * @return bool
+	 */
+	public static function has_products() {
```

Comment:
@ilyasfoo I think we could just use one query to get the same outcome. What do you think?

```php
public static function has_products() {
	$product_query = array(
		'post_type' => 'product',
		'post_status' => 'publish',
		'fields' => 'ids',
		'meta_query' => array(
			array(
				'key' => '_headstart_post',
				'compare' => 'NOT EXISTS'
			)
		)
	);
	$products = get_posts($product_query);
	$product_count = count($products);

	return $product_count > 0;
}
```

Reply:
Actually, you have a point here. `get_posts` can be set to return only the post ids, which means the data return size would be negligible.

In contrast, your solution would only require a single round trip to db which is better.  I'm leaning towards making the change